src/wp-admin/includes/export.php

```diff
@@ -243,6 +243,10 @@ function export_wp( $args = array() ) {
 	 * @return string
 	 */
 	function wxr_cdata( $str ) {
+		if ( ! is_string( $str ) ) {
+			return '';
+		}
```

Comment:
I believe I may be contradicting a previous suggestion I had here, but here's the latest informed by my [latest research](https://core.trac.wordpress.org/ticket/64347#comment:13):

```suggestion
		if ( ! is_string( $str ) && is_scalar( $str ) ) {
			$str = (string) $str;
		}
```

Reply:
Or:

```suggestion
		$str = (string) $str;
```

Reply:
Hi @westonruter, I guess the first suggestion should be something like below,

```php
if ( ! is_string( $str ) || is_scalar( $str ) ) {
    $str = (string) $str;
}
```

This is because, what if data is `null`? `! is_string( null )` would evaluate to `true`, but `is_scalar( null )` would evaluate to `false`, hence `$str` will remain as is and would not change to `string`, it would then gives the fatal for null value.

https://www.php.net/manual/en/function.is-scalar.php#:~:text=(PHP%204%20%3E=%204.0.,consider%20NULL%20to%20be%20scalar.

Reply:
I would like to go by casting it to `str` whatever value it is as per - https://core.trac.wordpress.org/ticket/64347#comment:14

Reply:
Oh, interesting. I wasn't aware that `is_scalar( null )` is `false`. That's news to me and counter-intuitive.